projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -784,12 +816,14 @@ export class IgxOverlayService implements OnDestroy {
         }
     };
 
-    private addOutsideClickListener(info: OverlayInfo) {
+    private addOutsideClickListener = (info: OverlayInfo) => {
         if (info.settings.closeOnOutsideClick) {
             if (info.settings.modal) {
                 fromEvent(info.elementRef.nativeElement.parentElement.parentElement, 'click')
-                    .pipe(takeUntil(this.destroy$))
```

Comment:
with `takeUntil(this.destroy$))` modal handlers were unsubscribed only when overlay service is destroyed - this actually did work, but subscriptions were removed too late.
with `takeUntil(this.removeClick$)` modal handlers will be unsubscribed once client calls `detach`. This is better because it will happen sooner. However what if client never calls `detach` - handlers will never be unsubscribed!

I think we can merge `destroy$` and `removeClick$` and use `takeUntil` on the merged result - this way we will ensure:
- subscription is cleared when `detach` is called.
- all subscriptions are cleared in `onDestroy`.

Finally - please move all this to another PR. We need to test this better and to see if we do not miss anything.